src/main/java/com/metamx/common/io/smoosh/SmooshedFileMapper.java

```diff
@@ -131,10 +132,20 @@ public ByteBuffer mapFile(String name) throws IOException
   }
 
   @Override
-  public void close() throws IOException
+  public void close()
   {
+    Throwable thrown = null;
     for (MappedByteBuffer mappedByteBuffer : buffersList) {
-      ByteBufferUtils.unmap(mappedByteBuffer);
+      try {
+        ByteBufferUtils.unmap(mappedByteBuffer);
+      } catch (Throwable t) {
+        if (thrown == null) {
+          thrown = t;
+        } else {
+          thrown.addSuppressed(t);
+        }
+      }
     }
+    Throwables.propagateIfPossible(thrown);
```

Comment:
This changes behavior as it will now completely swallow `IOException` instead of propagating it, or even logging it.

Is there a reason the exception should be swallowed now?